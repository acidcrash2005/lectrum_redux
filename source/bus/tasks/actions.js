// Types
import { types } from './types';

//Async
export const fetchTasksAsync = () => ({
    type: types.FETCH_TASKS_ASYNC,
});

export const createTasksAsync = (message) => ({
    type:    types.CREATE_TASKS_ASYNC,
    payload: message,
});

export const updateTasksAsync = (updateTask) => ({
    type:    types.UPDATE_TASKS_ASYNC,
    payload: updateTask,
});

export const removeTaskAsync = (idTask) => ({
    type:    types.REMOVE_TASKS_ASYNC,
    payload: idTask,
});

export const completeAllTasksAsync = (taskList) => ({
    type:    types.COMPLETE_ALL_TASKS_ASYNC,
    payload: taskList,
});

//Sync
export const fillTasks = (tasks) => {
    return {
        type:    types.FILL_TASKS,
        payload: tasks,
    };
};

export const createTasks = (task) => ({
    type:    types.CREATE_TASK,
    payload: task,
});

export const updateTasks = (updateTask) => ({
    type:    types.UPDATE_TASKS,
    payload: updateTask,
});

export const removeTask = (idTask) => ({
    type:    types.REMOVE_TASKS,
    payload: idTask,
});

export const completeAllTasks = (taskList) => ({
    type:    types.COMPLETE_ALL_TASKS,
    payload: taskList,
});

export const editTaskChange = (newMessage) => ({
    type:    types.EDIT_TASK_CHANGE,
    payload: newMessage,
});

export const editTaskFocus = (taskId) => ({
    type:    types.EDIT_TASK_FOCUS,
    payload: taskId,
});

