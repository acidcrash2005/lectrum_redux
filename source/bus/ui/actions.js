// Types
import { types } from './types';

export const startSpinner = () => {
    return {
        type: types.START_FETCHING,
    };
};

export const stopSpinner = () => {
    return {
        type: types.STOP_FETCHING,
    };
};

