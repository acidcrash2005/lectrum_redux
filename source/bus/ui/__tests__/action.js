//Actions
import * as actions from '../actions';

describe('ui actions:', () => {
    test('startSpinner', () => {
        expect(actions.startSpinner()).toMatchSnapshot();
    });

    test('stopSpinner', () => {
        expect(actions.stopSpinner()).toMatchInlineSnapshot(`
Object {
  "type": "STOP_FETCHING",
}
`);
    });


});
