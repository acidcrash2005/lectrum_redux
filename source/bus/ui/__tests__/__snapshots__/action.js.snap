// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui actions: startSpinner 1`] = `
Object {
  "type": "START_FETCHING",
}
`;
